include/picongpu/fields/MaxwellSolver/Yee/StencilFunctor.hpp

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2021 Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace maxwellSolver
+        {
+            namespace yee
+            {                
+                /** Base stencil functor to update fields inside the kernel
+                 *
+                 * This class serves to define the interface requirements for stencil functor implementations.
+                 * So if roughly defines a "concept".
```

Comment:
Here may add "Not intended for polymorphic use", although no virtual destructor already means that in C++